src/cgi/wwsympa.fcgi.in

```diff
@@ -7313,6 +7314,79 @@ sub do_auth_del {
 
     return ($in{'previous_action'} || 'sigindex');
 }
+# Deletes user from lists (requested by listmaster)
+sub do_mass_del {
+    wwslog('info', '(%s) (%s)', $in{'email'},
+        join(', ', split /\0/, $in{'lists'}));
+
+    # Access control.
+    return undef
+        unless (Sympa::is_listmaster($robot, $param->{'user'}{'email'}));
+
+    # Turn data into usable structures
```

Comment:
All inputs coming form outside of program should be checked their validity.  `@lists` shouldn't contain non-existing lists and `$email` should be a valid email address.

Reply:
`$param->{'user'}{'email'}` is provided by Sympa, isn’t it? Isn’t it the email of the current user?
We use `$param->{'user'}{'email'}` directly in numerous functions of `src/cgi/wwsympa.fcgi.in`.

Reply:
Since I [now use](https://github.com/sympa-community/sympa/pull/605#discussion_r285056421) %required_privileges for access control, I removed that part.

Reply:
How can I check that a list exists?

Reply:
Such as
```perl
next unless Sympa::List->new($list, $robot, {just_try => 1});
```

Reply:
Can you check d19ca6aa2c75cb5376cb21b7d9c986797614f172 to see if it suits you, please?

Reply:
I'm ok.

I missed another thing.
```
    $email = Sympa::Tools::Text::valid_email($email);

    return $in{'previous_action'} || 'serveradmin' unless $email;
```
should be
```
    return $in{'previous_action'} || 'serveradmin' unless Sympa::Tools::Text::valid_email($email);
```

Reply:
Fixed in a9a7ff6fc (I rebased my previous commit).

Reply:
Thanks! I'm ok.